homeassistant/auth/__init__.py

```diff
@@ -453,6 +457,36 @@ def async_create_access_token(
             algorithm="HS256",
         ).decode()
 
+    @callback
+    def _async_resolve_provider(
+        self, refresh_token: models.RefreshToken
+    ) -> Optional[AuthProvider]:
+        """Get the auth provider for the given refresh token.
+
+        Raises an exception if the expected provider is no longer available or return
+        None if no provider was expected for this refresh token.
+        """
+        if not refresh_token.cred or refresh_token.cred.auth_provider_id is None:
+            return None
```

Comment:
We should start storing the Home Assistant version that was used to generate the refresh token. We can then also raise an error if we find a refresh token that is not system generated but has no provider.

Reply:
The version have now been added to the created tokens. I did not add the check now since I have no way of known when this get's merged. I can post a separate patch with a check once this one is in.